### ISSUE TITLE
Makefile: Change how JavaScript building works, set minimum spec to ES2017 (async)

### ISSUE THREAD

**Issue by mikekasprzak** [JammerCore/JammerCore#2078]
Like #2076, change how JavaScript files are built.

Modern tools like [ESBuild](https://esbuild.github.io/) (Go) and [SWC](https://swc.rs/) (Rust) seem to be in an arms race for features and performance (I mean the SWC developer semes to think he's in one :rofl:). We should support both tools, and try to keep both pipelines working, assuming one doesn't have a _killer_ feature we can't live without.

Both tools _can_ be used as a replacement for Buble, but it's been suggested that full builds take seconds (max) with these tools.

## Prerequisites
Before we can switch away from Buble and the current builds, we need to stop using non-modules and hackery to access fucnctios

* [ ] No more `window.node_Blah` (`node_Blah`) functions in the global scope
* [ ] No more global scope Prism highlighter
* [ ] No more global scope Markdown parser
* [ ] No more global scope polyfills
  * [ ] Stretch goal: new polyfill solution (I want `String.at()`). Something like https://github.com/zloirock/core-js

## Add support for ESBuild to Makefile
* [ ] ESBuild support

## Add support for SWC to Makefile
* [ ] SWC support?
